### ISSUE TITLE
TGeo visibility level control

### ISSUE THREAD

**Issue by c-dilks** [root-project/jsroot#258]
I'm having trouble controlling the `TGeo` visibility level using the `vislvlN` option in a URL. 

I'm testing with the following ROOT file (tarballed for github) for the EIC dRICH:
[detector_geometry.root.tar.gz](https://github.com/root-project/jsroot/files/11738419/detector_geometry.root.tar.gz)

With a self-hosted instance of `jsroot` version 7.3.4, I have tried the following URLs:

Default visiblity level:
```
http://localhost:8000/?file=detector_geometry.root&item=default/world_volume/DRICH_0/DRICH_gas_0
```

Increase visibility level to 8:
```
http://localhost:8000/?file=detector_geometry.root&item=default/world_volume/DRICH_0/DRICH_gas_0&opt=vislvl8
```

In both cases, I see the same result:
![Screenshot 2023-06-13 at 15-16-35 Read a ROOT file](https://github.com/root-project/jsroot/assets/7843751/b667ddba-b349-4d74-8650-95c03666f60e)

With the increased visibility level, I expect to see the sensors, as in this screenshot from ROOT `ogl` viewer:
![2023-06-13-151908_608x624_scrot](https://github.com/root-project/jsroot/assets/7843751/96ea29ff-bd68-4749-a5b9-88c2c1b3e619)

I think the sensor volumes are one visibility level higher than ROOT's default, since I need to increase the visibility level in ROOT `ogl` to see them.

Is the `vislvlN` URL option not working correctly, or is my syntax wrong? Would it be possible to add a visibility level control to the `Controls` box?

**Comment by linev** [comment]
Hi,

I can reproduce problem, which has several aspects.

It is not only visibility level. For the geometry it only plays a role when top volume (or geo manager itself) displayed.
In such case `fVisLevel=3` which is set in `TGeoManager`, really prevents additional nodes from rendering.
But when you specify sub-volume like `default/world_volume/DRICH_0/DRICH_gas_0`, level=3 would be ok.

But there is limitation in JSROOT for number of physical nodes which are rendered. 
Up to now it was 10000 nodes - and full rendering of the geometry includes ~15000 physical nodes.
This is also default number in `TGeoManager`. The only way to change this limit up to now - call `gGeoManager->SetMaxVisNodes(20000)` before storing into the ROOT file. You can try.

Today I add feature to increase this number by providing `opt=more` or `opt=more3` to draw option in URL string.
This should work with current master and `7.3.` branch. The URL should look like:

```
http://localhost:8000/?file=detector_geometry.root&item=default&opt=vislvl4,more
```

I will add these parameters also to control GUI.

Regards,
Sergey

**Comment by linev** [comment]
I also add selection to control GUI.

![geo_ctrl](https://github.com/root-project/jsroot/assets/4936580/b30fccbb-9f40-4266-9095-e0b6a7bd41a8)

You can try it.
Current master will be published very soon - probably today as `7.4.0` version.

Thanks for reporting the problem

**Comment by c-dilks** [comment]
Very nice, thank you so much!
